### PR TITLE
wl-gammactl: init at unstable-2021-09-13

### PR DESCRIPTION
**Pull request by lodi**
###### Description of changes
Packaged mischw/[wl-gammactl](https://github.com/mischw/wl-gammactl) for contrast/brightness/gamma adjustment on Wayland compositors.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/wayland/wl-gammactl/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{ lib, stdenv, fetchFromGitHub
+, meson, pkg-config, ninja, git
+, wayland, wayland-scanner, wlroots, wayland-protocols, gtk3
+}:
+
+stdenv.mkDerivation rec {
+  pname = "wl-gammactl";
+  version = "master-20210913";
```
Comment (lovesegfault) [inline pkgs/tools/wayland/wl-gammactl/default.nix:8]:
```suggestion
  version = "unstable-2021-09-13";
```

File: pkgs/tools/wayland/wl-gammactl/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  src = fetchFromGitHub {
+    owner = "mischw";
+    repo = pname;
+    rev = "e2385950d97a3baf1b6e2f064dd419ccec179586";
+    sha256 = "4ve38pMGKDDC4oZZERyPdxVJX4RgHwYqX6IUQQ2GpGg=";
+    fetchSubmodules = true;
```
Comment (lovesegfault) [inline pkgs/tools/wayland/wl-gammactl/default.nix:15]: It seems this is only necessary because upstream has opted to vendor in `wlr-protocols`. I don't think that's needed, you can just take them from wlroots at build-time?

File: pkgs/tools/wayland/wl-gammactl/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  meta = with lib; {
+    description = "Contrast, brightness, and gamma adjustments for Wayland";
+    longDescription = ''
+      Small GTK GUI application to set contrast, brightness, and gamma for wayland compositors which
+      support the wlr-gamma-control protocol extension.
+    '';
+    homepage = "https://github.com/mischw/wl-gammactl";
+    license = licenses.mit;
+    platforms = platforms.unix;
```
Comment (lovesegfault) [inline pkgs/tools/wayland/wl-gammactl/default.nix:29]: I imagine this only works on Linux, given it's aimed at wayland?

**Review by lovesegfault — CHANGES_REQUESTED** [review]

File: pkgs/tools/wayland/wl-gammactl/default.nix (hunk 2 above)
Comment (lodi) [reply pkgs/tools/wayland/wl-gammactl/default.nix]: Thanks @lovesegfault, I made the changes you suggested.

**Comment by lovesegfault** [comment]
Result of `nixpkgs-review pr 163388` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wl-gammactl</li>
  </ul>
</details>

**Review by lovesegfault — APPROVED** [review]
This looks good, can you rebase your commits into a single one?

`wl-gammactl: init at unstable-2021-09-12`

Thanks!

**Comment by lodi** [comment]
@lovesegfault Sure, done.

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -7217,6 +7217,12 @@
     githubId = 1267527;
     name = "Daniel Firth";
   };
+  lodi = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:7220]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

File: pkgs/tools/wayland/wl-gammactl/default.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  version = "unstable-2021-09-13";
+
+  src = fetchFromGitHub {
+    owner = "mischw";
+    repo = pname;
+    rev = "e2385950d97a3baf1b6e2f064dd419ccec179586";
+    sha256 = "8iMJK4O/sNIGPOBZQEfK47K6OjT6sxYFe19O2r/VSr8=";
+  };
+
+  nativeBuildInputs = [ meson pkg-config ninja git ];
```
Comment (SuperSandro2000) [inline pkgs/tools/wayland/wl-gammactl/default.nix:17]: Why do we need git? It most likely needs special handling.

File: pkgs/tools/wayland/wl-gammactl/default.nix [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  buildInputs = [ wayland wlroots gtk3 ];
+
+  postUnpack = "cp -r ${wlr-protocols}/share/wlr-protocols source";
```
Comment (SuperSandro2000) [inline pkgs/tools/wayland/wl-gammactl/default.nix:20]:
```suggestion
  postUnpack = ''
    cp -r ${wlr-protocols}/share/wlr-protocols source
  '';
```
could we use `ln -s`?

File: maintainers/maintainer-list.nix (hunk 4 above)
Comment (lodi) [reply maintainers/maintainer-list.nix:7220]: Okay, I split it into its own commit.  I tried to insert myself as alphabetically as possible but the file is already somewhat loosely sorted.

File: pkgs/tools/wayland/wl-gammactl/default.nix (hunk 5 above)
Comment (lodi) [reply pkgs/tools/wayland/wl-gammactl/default.nix]: There's a build step in the meson script that fetches the git submodule automatically but I patched it out.

File: pkgs/tools/wayland/wl-gammactl/default.nix [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+  nativeBuildInputs = [ meson pkg-config ninja ];
+  buildInputs = [ wayland wlroots gtk3 ];
+
+  postUnpack = ''
+    rmdir source/wlr-protocols
+    ln -s ${wlr-protocols}/share/wlr-protocols source
+  '';
+
+  patchPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/tools/wayland/wl-gammactl/default.nix:25]:
```suggestion
  postPatch = ''
```
